app/main/views/frameworks.py

```diff
@@ -230,7 +230,7 @@ def get_framework_suppliers(framework_slug):
         agreement_returned = request.args.get('agreement_returned')
         if agreement_returned is not None:
             if convert_to_boolean(agreement_returned):
-                requested_statuses = requested_statuses or ('signed', 'on-hold', 'countersigned')
+                requested_statuses = requested_statuses or ('signed', 'on-hold', 'approved', 'countersigned')
```

Comment:
Agreed - there is an API "tidy-up" to be done once we can safely ignore draft agreements everywhere and I'll address at least some of the python-vs-sql-ness then.